blockchain/backup/NFTtheWorld.sol

```diff
@@ -108,56 +112,65 @@ contract NFTtheWorld {
     // During the execution all joined users get their previously specificed number of NFTs randomly assigned.
     // To make sure no NFT gets assigned to multiple users, it is removed from the list of available NFTs.
     function drop(uint256 _dropHash) public {
-        //TODO: make sure that only the drop creator can execute the drop
         require(
             nftOwnerships[_dropHash][0].dropTime <= block.timestamp,
             "Droptime not yet reached!"
         );
-        uint256 nftElement;
         for (uint256 i = 0; i < joinedUsers[_dropHash].length; i++) {
+            shuffle(_dropHash);
+            // loop trough the number of NFTs j reserved by user i
             for (
                 uint256 j = 0;
                 j < nftReservations[joinedUsers[_dropHash][i]][_dropHash];
                 j++
             ) {
-                nftElement =
-                    generateRandomNumber(j) %
-                    availableNFTs[_dropHash].length;
-                remove(nftElement, _dropHash);
+                uint256 nftElement = getNFTIndex(
+                    availableNFTs[_dropHash][j],
+                    _dropHash
+                );
                 nftOwnerships[_dropHash][nftElement].reservedFor = joinedUsers[
                     _dropHash
                 ][i];
-                nftReservationInformationOfUsers[joinedUsers[_dropHash][i]]
-                    .push(nftOwnerships[_dropHash][nftElement].nftId);
+                nftReservationInformationOfUsers[joinedUsers[_dropHash][i]][
```

Comment:
Haha, I notice it becomes readable when reading it from right to left: `nftDrop.joinedUsers[i].nftReservationInfo`.